packages/@sanity/schema/src/legacy/types/slug.js

```diff
@@ -46,9 +46,9 @@ const SLUG_CORE = {
       }
 
       const errorMessage = 'Slug is already in use'
-      const checkUnique = get(options, 'type.options.checkUnique', defaultChecker)
-      return Promise.resolve(checkUnique(value.current, options)).then(
-        hasDupe => (hasDupe ? errorMessage : true)
+      const isUnique = get(options, 'type.options.isUnique', defaultUniqueCheck)
```

Comment:
Hm... maybe it should have been named `defaultIsUnique` instead of `defaultUniqueCheck`?